### PR TITLE
Confirm file exists before setting it to $requested_file - #5268

### PR DESCRIPTION
**Pull request by pippinsplugins**
#5268

### CODE REVIEW

File: includes/process-download.php [hunk 1]
```diff
@@ -838,3 +850,30 @@ function edd_set_requested_file_scheme( $requested_file, $download_files, $file_
...
+ * @return void
+ */
+function edd_check_file_url_head( $requested_file, $args, $method ) {
+
+	// If this is a file URL (not a path), perform a head request to determine if it's valid
+	if( filter_var( $requested_file, FILTER_VALIDATE_URL ) ) {
+
+		$request = wp_remote_head( $requested_file );
+
+		if( is_wp_error( $request) ) {
```
Comment (cklosowski) [inline includes/process-download.php]: I don't believe that a `404` returns `WP_Error` but requires checking the `wp_remote_retrieve_response_code`.

Should we do something like this instead?
`( is_wp_error( $request ) || (int) wp_remote_retrieve_response_code( $request ) === 404 )`

**Comment by cklosowski** [comment]
Doh, looks like we need to update something in our unit tests for `latest`. I'll rerun the `latest` tests for WP Core and see if that fixes it, and we were just in a weird state for the release cycle.

File: includes/process-download.php (hunk 1 above)
Comment (pippinsplugins) [inline includes/process-download.php]: I was having a hard time triggering a 404 error code. It was strange, even files that were clearly 404'ing were returning a 200 code with wp_remote_head(). No idea why.

Checking for a 404 code does seem like a good idea.

Comment (pippinsplugins) [inline includes/process-download.php]: @cklosowski Added 404 check.

**Pull request merged**